src/common/types/uuid.cpp

```diff
@@ -0,0 +1,89 @@
+#include "common/types/uuid.h"
+
+namespace kuzu {
+namespace common {
+
+bool UUID::fromString(std::string str, int128_t& result) {
+    auto hex2Char = [](char ch) -> unsigned char {
```

Comment:
Instead of using lambda, better to make hex2Char and isHex as static functions in hte UUID class